psutil/tests/__init__.py

```diff
@@ -57,11 +58,6 @@
 from psutil._compat import which
 
 
-if PY3:
-    import unittest
-else:
-    import unittest2 as unittest  # requires "pip install unittest2"
```

Comment:
I'm so happy to see this go! Other test files (`test_linux.py`) should stop doing:

```
from psutil.tests import unittest
```

...and instead do:

```
import unittest
```

Could you please update those files as well?

Reply:
This file still has a monkey patch of `unittest.TestCase` for a modified `__str__`, `assertRaisesRegex` compat code for 2.7 and handling for `multiprocessing.Pool`.

Reply:
I think the monkey patch should still work because `test/__init__.py` is imported before other `test/*.py` files.
When they do `import unittest` the should automatically inherit the monkey-patched version of `unittest.TestCase`.

Reply:
Hmm, https://github.com/giampaolo/psutil/pull/2099/commits/4bc28e515b5ba344a657ea302eea281415c0543f passed on GHA but failed on AppVeyor:

```
psutil.tests.test_windows.TestSystemAPIs.test_nic_names ... ok
psutil.tests.test_windows.TestSystemAPIs.test_pids ... skipped 'test not relieable on appveyor'
psutil.tests.test_windows.TestSystemAPIs.test_total_phymem ... ok
psutil.tests.test_windows.WindowsTestCase.runTest ... ok
======================================================================
ERROR: psutil.tests.test_process.TestProcess.test_nice
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\psutil\psutil\tests\test_process.py", line 812, in test_nice
    with self.subTest(prio=prio):
AttributeError: 'TestProcess' object has no attribute 'subTest'
----------------------------------------------------------------------
Ran 575 tests in 12.912s
FAILED (errors=1, skipped=263)
FAILED
Command exited with code 1
```

Reply:
I guess `unittest2` package provided `subTest` method on Python 2.7, a unittest method which only appeared later in python 3.X. Not a big deal. I prefer to get rid of unittest2 dependency and fake `subTest` method existence in our `unittest.TestCase` subclass. I think something like this should work:

```python
if not PY3:
    @contextlib.contextmanager
    def subTest(self, *args, **kw):
        # fake it for python 2.7
        yield
```

Reply:
That looks better, thanks!

(Another option is dropping 2.7 ;)

Reply:
I think 2.7 will stay for a very long time (I'd put a but to at least one decade).